jsl-tinylog/src/main/java/org/tinylog/jsl/TinylogLogger.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2021 Gerrit Rode
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.tinylog.jsl;
+
+import java.util.Objects;
+import java.util.ResourceBundle;
+import java.util.function.Supplier;
+
+import org.tinylog.configuration.Configuration;
+import org.tinylog.format.JavaTextMessageFormatFormatter;
+import org.tinylog.format.MessageFormatter;
+import org.tinylog.provider.LoggingProvider;
+import org.tinylog.provider.ProviderRegistry;
+
+/**
+ * Java System.Logger compatible logger that uses tinylog's {@link LoggingProvider}.
+ */
+public class TinylogLogger implements System.Logger {
+
+	private static final int STACKTRACE_DEPTH = 2;
+
+	private static final MessageFormatter MESSAGE_FORMATTER = new JavaTextMessageFormatFormatter(Configuration.getLocale());
+	private static final LoggingProvider LOGGING_PROVIDER = ProviderRegistry.getLoggingProvider();
+
+	private final String name;
+
+	/**
+	 * @param name   the name of the logger
+	 */
+	public TinylogLogger(final String name) {
+		this.name = Objects.requireNonNull(name);
+	}
+
+	@Override
+	public String getName() {
+		return name;
+	}
+
+	@Override
+	public boolean isLoggable(final Level level) {
+		Objects.requireNonNull(level);
+
+		// Setting the log level of a message to OFF makes little sense and should result in the message not being logged.
+		if (level == Level.OFF) {
+			return false;
+		} else {
+			return LOGGING_PROVIDER.isEnabled(STACKTRACE_DEPTH, null, translate(level));
+		}
+	}
+
+	@Override
+	public void log(final Level level, final String msg) {
+		Objects.requireNonNull(level);
+
+		if (isLoggable(level)) {
```

Comment:
I have just noticed that we have a bug here. This call of `isLoggable()` adds an extra element to the stack trace and the stack trace depth of 2 won't be correct anymore. I assume we need a stack trace of 3 inside of `isLoggable()` in this case. I propose either to replace `isLoggable()` by `level != Level.OFF && LOGGING_PROVIDER.isEnabled(STACKTRACE_DEPTH, null, translate(level))` or to add another internal `isLoggable()` method that passes a stack trace depth of 3 to the logging provider.

Reply:
Thanks, I will look at it again next weekend ;)